yt/python/yt/wrapper/query_commands.py

```diff
@@ -195,3 +204,303 @@ def format_time(time):
+        if new_state.is_running():
+            new_progress_line = self.get_running_progress_line()
+            if self.progress_line != new_progress_line:
+                self.log("query %s: %s", self.query.id, new_progress_line)
+            self.progress_line = new_progress_line
+        elif new_state != self.state:
+            self.log("query %s %s", self.query.id, new_state)
+        self.state = new_state
+
+    # This method is to be overridden in subclasses for concrete engines.
+    def get_running_progress_line(self):
+        """Return a line that is to be printed for a running operation. It is printed only
+        if it differs from its previous state"""
+        return "running"
+
+    def log(self, message, *args, **kwargs):
+        elapsed_seconds = (utcnow() - self.start_time).total_seconds()
+        message = "({0:2} min) ".format(int(elapsed_seconds) // 60) + message
+        logger.log(self.level, message, *args, **kwargs)
+
+
+class YqlQueryInfoPrinter(GenericQueryInfoPrinter):
+    def get_running_progress_line(self):
+        progress = self.query.get_progress()
+        yql_progress = progress.get("yql_progress", {})
+        # YQL node progress contains something similar to YT progress counter and also some additional
+        # fields like "finishedAt" or "stages". We want to aggregate numbers that are similar to YT progress
+        # and output them similar to how YT operation progress is printed.
+        key_whitelist = ["aborted", "completed", "failed", "lost", "pending", "running", "total"]
+        total_progress = defaultdict(int)
+        for node_id, node_progress in yql_progress.items():
+            for key, value in node_progress.items():
+                if key in key_whitelist:
+                    total_progress[key] += value
+        return " ".join("{0}={1:<5}".format(k, v) for k, v in order_progress(total_progress))
+
+
+class Query(object):
+    """Holds information about query."""
+    def __init__(self, id, engine, query_tracker_stage=None,
+                 abort_exceptions=(KeyboardInterrupt, TimeoutError), client=None):
+        self.id = id
+        self.engine = engine
+        self.abort_exceptions = abort_exceptions
+        self.client = client
+        self.url = get_query_url(id, client=client)
+        self.query_tracker_stage = query_tracker_stage
+
+        # TODO(max42): CHYT support was introduced quite recently, so we do not support it yet. QL does not
+        # provide means of tracking query progress, so we do not support it either. SPYT will also follow later.
+        printers = {
+            "yql": YqlQueryInfoPrinter,
+        }
+        self.printer = printers.get(self.engine, GenericQueryInfoPrinter)(self, client=client)
+
+    def abort(self, message=None):
+        """Aborts query."""
+        abort_query(self.id, message=message, stage=self.query_tracker_stage, client=self.client)
+
+    def get_state_monitor(self, time_watcher):
```

Comment:
If you decide to rename `get_query_state_monitor` this method should be renamed as well.